src/main/java/org/mockito/internal/creation/bytebuddy/SubclassBytecodeGenerator.java

```diff
@@ -185,7 +185,13 @@ public <T> Class<? extends T> mockClass(MockFeatures<T> features) {
                             + features.mockedType.getSimpleName();
         }
         String name =
-                String.format("%s$%s$%d", typeName, "MockitoMock", Math.abs(random.nextInt()));
+                String.format(
+                        "%s$%s$%s",
+                        typeName,
+                        "MockitoMock",
+                        GraalImageCode.getCurrent().isDefined()
```

Comment:
Is there a specific reason we only want this on Graal? I can imagine we would want this for others as well, to keep everything consistent?

Reply:
The very theoretical situation would be a OSGi setup where multiple modules of Mockito wanted to create a mock for the same class loader. It's unlikely but might trigger a regression. The question is if the stable name justifies this theoretical issue.